<a href="https://colab.research.google.com/github/oscarstivenmoreno2008-coder/Porgramacion/blob/main/quiz%203-26%20de%20mrzo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Quiz 3
26 de marzo
## ***Oscar*** Moreno 2261347

In [35]:
%%writefile promedio.py
import statistics
def calcular_promedio(n1, n2, n3):
    return (n1 + n2 + n3) / 3

def mediana(n1, n2, n3):
  return statistics.median([n1, n2, n3])

def moda(n1, n2, n3):
  return statistics.mode([n1, n2, n3])

Overwriting promedio.py


In [16]:
%%writefile pedirnotas.py
def pedir_nombre():
    return input("ingrese el nombre del estudiate: ")

def pedir_notas():
    n1 = float(input("nota 1: "))
    n2 = float(input("nota 2: "))
    n3 = float(input("nota 3: "))
    n4 = float(input("nota 4: "))
    return [n1, n2, n3, n4]

Writing pedirnotas.py


In [37]:
import pedirnotas
pedirnotas.pedir_nombre()

ingrese el nombre del estudiate: Oscar Moreno


'Oscar Moreno'

In [38]:
notas = pedirnotas.pedir_notas()

nota 1: 2
nota 2: 3
nota 3: 4
nota 4: 1


In [39]:
notas

[2.0, 3.0, 4.0, 1.0]

In [23]:
promedio_notas = sum(notas) / len(notas)
print(f"El promedio de las notas es: {promedio_notas}")

El promedio de las notas es: 5.0


In [47]:
import statistics

mediana_notas = statistics.median(notas)
print(f"La mediana de las notas es: {mediana_notas}")

La mediana de las notas es: 2.5


In [49]:
import promedio
import importlib
importlib.reload(promedio)

moda_notas = promedio.moda(notas[0], notas[1], notas[2])
print(f"La moda de las notas es: {moda_notas}")

La moda de las notas es: 2.0


In [50]:
if promedio_notas >= 3.0:
    print("El estudiante aprobó.")
else:
    print("El estudiante reprobó.")

El estudiante aprobó.


Para las notas ingresadas, el promedio es 5.0, la mediana es 2.5 y la moda es 2.0 (calculada con las tres primeras notas). El estudiante ha aprobado ya que su promedio es mayor o igual a 3.0.